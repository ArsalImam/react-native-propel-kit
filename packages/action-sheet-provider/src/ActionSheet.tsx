import React, {forwardRef, RefForwardingComponent} from 'react';
import {
  Dimensions,
  Platform,
  StyleSheet,
  StyleProp,
  ViewStyle,
  ActionSheetIOSOptions,
  TextStyle,
  ScrollView,
  ScrollViewProps
} from 'react-native';
import ModalDialog, {ModalDialogProps, ModalDialogHandle} from '@mgcrea/react-native-modal-dialog';
import Button from '@mgcrea/react-native-button';

import isUndefined from './utils/isUndefined';

export type Props = Pick<ScrollViewProps, 'scrollEnabled'> &
  Omit<ModalDialogProps, 'onConfirm' | 'confirmTitle' | 'confirmStyle'> &
  ActionSheetIOSOptions & {
    optionStyle?: StyleProp<TextStyle>;
    lastOptionExtraStyle?: StyleProp<TextStyle>;
    destructiveButtonColor?: string;
    onButtonPress: (buttonIndex: number) => void;
    onCancel: () => void;
  };

const MAX_HEIGHT = Dimensions.get('window').height;

export const defaultProps = {
  backgroundColor: 'white',
  cancelTitle: 'Cancel',
  destructiveButtonColor: Platform.select({
    ios: '#ff3b2f',
    android: '#ff3b2f'
  }),
  containerStyle: Platform.select<ViewStyle>({
    ios: {marginHorizontal: 8, maxHeight: MAX_HEIGHT - 8},
    android: {}
  }),
  headerStyle: Platform.select<ViewStyle>({
    ios: {alignItems: 'center', padding: 16, borderTopLeftRadius: 12, borderTopRightRadius: 12, opacity: 0.94},
    android: {alignItems: 'flex-start', padding: 16}
  }),
  bodyStyle: Platform.select<ViewStyle>({
    ios: {
      flexShrink: 1,
      flexDirection: 'column',
      backgroundColor: 'transparent',
      opacity: 0.94
    },
    android: {
      flexShrink: 1,
      flexDirection: 'column',
      backgroundColor: 'transparent'
    }
  }),
  footerStyle: Platform.select<TextStyle>({
    ios: {flex: 0, backgroundColor: 'transparent', marginVertical: 8, borderRadius: 12},
    android: {}
  }),
  titleStyle: Platform.select<TextStyle>({
    ios: {paddingBottom: 10, fontSize: 14, fontWeight: '500', textAlign: 'center', color: '#888'},
    android: {fontSize: 20, fontWeight: '500', color: '#333'}
  }),
  messageStyle: Platform.select<TextStyle>({
    ios: {paddingBottom: 8, fontSize: 13, fontWeight: '400', textAlign: 'center', color: '#888'},
    android: {fontSize: 20, fontWeight: '400', color: '#333'}
  }),
  cancelStyle: Platform.select<TextStyle>({
    ios: {fontSize: 20, color: '#007aff', paddingVertical: 5, borderRadius: 12, fontWeight: '600'},
    android: {}
  }),
  optionStyle: Platform.select<TextStyle>({
    ios: {
      fontSize: 20,
      color: '#007aff',
      paddingBottom: 5,
      paddingTop: 5 - StyleSheet.hairlineWidth,
      marginTop: StyleSheet.hairlineWidth,
      borderRadius: 0
    },
    android: {fontSize: 20}
  }),
  lastOptionExtraStyle: Platform.select<TextStyle>({
    ios: {
      borderBottomLeftRadius: 12,
      borderBottomRightRadius: 12
    },
    android: {fontSize: 20}
  })
};

const ActionSheet: RefForwardingComponent<ModalDialogHandle, Props> = (
  {
    options,
    backgroundColor = defaultProps.backgroundColor,
    title,
    titleStyle = defaultProps.titleStyle,
    message,
    messageStyle = defaultProps.messageStyle,
    optionStyle = defaultProps.optionStyle,
    containerStyle = defaultProps.containerStyle,
    lastOptionExtraStyle = defaultProps.lastOptionExtraStyle,
    headerStyle = defaultProps.headerStyle,
    bodyStyle = defaultProps.bodyStyle,
    footerStyle = defaultProps.footerStyle,
    cancelStyle = defaultProps.cancelStyle,
    cancelButtonIndex,
    destructiveButtonIndex,
    destructiveButtonColor = defaultProps.destructiveButtonColor,
    onButtonPress,
    onCancel,
    scrollEnabled: propScrollEnabled,
    ...otherModalProps
  },
  ref
) => {
  const cancelTitle = !isUndefined(cancelButtonIndex) && options[cancelButtonIndex] ? options[cancelButtonIndex] : '';
  const scrollEnabled = !isUndefined(cancelButtonIndex) ? options.length > 6 : propScrollEnabled;

  return (
    <ModalDialog
      {...{
        ref,
        backgroundColor,
        title,
        titleStyle,
        message,
        containerStyle,
        messageStyle,
        onCancel,
        cancelTitle,
        headerStyle,
        bodyStyle,
        footerStyle,
        cancelStyle
      }}
      {...otherModalProps}
    >
      <ScrollView scrollEnabled={scrollEnabled}>
        {options.map((buttonTitle, buttonIndex) => {
          // Skip cancel button
          if (!isUndefined(cancelButtonIndex) && buttonIndex === cancelButtonIndex) {
            return null;
          }

          const color: string =
            destructiveButtonIndex && buttonIndex === destructiveButtonIndex ? destructiveButtonColor : '';
          return (
            <Button
              style={[optionStyle].concat(
                color ? {color} : false,
                buttonIndex === options.length - 1 ? lastOptionExtraStyle : false
              )}
              onPress={() => onButtonPress(buttonIndex)}
              title={buttonTitle}
              key={`${buttonIndex}`} // eslint-disable-line react/no-array-index-key
            />
          );
        })}
      </ScrollView>
    </ModalDialog>
  );
};

export default forwardRef(ActionSheet);
