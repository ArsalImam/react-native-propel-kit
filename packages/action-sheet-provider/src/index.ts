import ActionSheetProvider, {
  ActionSheetContext,
  Props as ActionSheetProviderProps,
  ContextProps as ActionSheetContextProps
} from './ActionSheetProvider';

export default ActionSheetProvider;
export {ActionSheetProvider, ActionSheetProviderProps, ActionSheetContext, ActionSheetContextProps};
