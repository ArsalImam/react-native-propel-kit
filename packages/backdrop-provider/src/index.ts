import BackdropProvider, {
  BackdropContext,
  Props as BackdropProviderProps,
  ContextProps as BackdropContextProps,
  defaultProps as backdropProviderDefaultProps
} from './BackdropProvider';

export default BackdropProvider;
export {BackdropProvider, BackdropProviderProps, backdropProviderDefaultProps, BackdropContext, BackdropContextProps};
