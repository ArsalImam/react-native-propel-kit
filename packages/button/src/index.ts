import Pressable, {
  Props as PressableProps,
  defaultProps as pressableDefaultProps,
  defaultStyles as pressableDefaultStyles
} from './Pressable';

import InputButton, {
  Props as InputButtonProps,
  defaultProps as inputButtonDefaultProps,
  defaultStyles as inputButtonDefaultStyles
} from './InputButton';

import Button, {
  Props as ButtonProps,
  defaultProps as buttonDefaultProps,
  defaultStyles as buttonDefaultStyles
} from './Button';

export default Button;

export {
  Pressable,
  PressableProps,
  pressableDefaultProps,
  pressableDefaultStyles,
  Button,
  ButtonProps,
  buttonDefaultProps,
  buttonDefaultStyles,
  InputButton,
  InputButtonProps,
  inputButtonDefaultProps,
  inputButtonDefaultStyles
};
