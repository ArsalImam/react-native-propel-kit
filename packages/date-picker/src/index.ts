import DatePicker, {Props as DatePickerProps, Handle as DatePickerHandle} from './DatePicker';
import TimePicker, {Props as TimePickerProps} from './TimePicker';
import MonthPicker, {Props as MonthPickerProps} from './MonthPicker';
import YearPicker, {Props as YearPickerProps} from './YearPicker';
import DateTimePicker, {Props as DateTimePickerProps} from './DateTimePicker';

export default DatePicker;
export {
  DatePicker,
  DatePickerProps,
  DatePickerHandle,
  TimePicker,
  TimePickerProps,
  DateTimePicker,
  DateTimePickerProps,
  MonthPicker,
  MonthPickerProps,
  YearPicker,
  YearPickerProps
};
