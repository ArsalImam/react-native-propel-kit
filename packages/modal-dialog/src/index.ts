import ModalDialog, {Props as ModalDialogProps, Handle as ModalDialogHandle} from './ModalDialog';

export default ModalDialog;
export {ModalDialog, ModalDialogProps, ModalDialogHandle};
