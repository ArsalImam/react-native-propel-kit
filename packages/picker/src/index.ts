import Picker, {Props as PickerProps, Item as PickerItem} from './Picker';

export default Picker;
export {Picker, PickerProps, PickerItem};
