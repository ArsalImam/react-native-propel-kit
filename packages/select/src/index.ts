import Select, {Props as SelectProps, Item as SelectItem} from './Select';

export default Select;
export {SelectProps, SelectItem};
