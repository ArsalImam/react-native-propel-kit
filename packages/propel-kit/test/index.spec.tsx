import React from 'react';
import 'react-native';

import * as exportedInterface from '../src';

describe('propel-kit', () => {
  it('export a proper interface', () => {
    expect(exportedInterface).toMatchSnapshot();
  });
});
