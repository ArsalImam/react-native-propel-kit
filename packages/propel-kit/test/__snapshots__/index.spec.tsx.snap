// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propel-kit export a proper interface 1`] = `
Object {
  "ActionSheetContext": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": null,
    "_currentValue2": null,
    "_threadCount": 0,
  },
  "ActionSheetContextProps": undefined,
  "ActionSheetProvider": [Function],
  "ActionSheetProviderProps": undefined,
  "BackdropContext": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": null,
    "_currentValue2": null,
    "_threadCount": 0,
  },
  "BackdropContextProps": undefined,
  "BackdropProvider": [Function],
  "BackdropProviderProps": undefined,
  "Button": [Function],
  "ButtonProps": undefined,
  "DatePicker": Object {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "DatePickerHandle": undefined,
  "DatePickerProps": undefined,
  "DateTimePicker": [Function],
  "DateTimePickerProps": undefined,
  "InputButton": [Function],
  "InputButtonProps": undefined,
  "ModalDialog": Object {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "ModalDialogHandle": undefined,
  "ModalDialogProps": undefined,
  "MonthPicker": Object {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "MonthPickerProps": undefined,
  "Picker": [Function],
  "PickerItem": [Function],
  "PickerProps": undefined,
  "Pressable": [Function],
  "PressableProps": undefined,
  "Select": Object {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "SelectItem": [Function],
  "SelectProps": undefined,
  "TimePicker": [Function],
  "TimePickerProps": undefined,
  "YearPicker": Object {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "YearPickerProps": undefined,
  "backdropProviderDefaultProps": Object {
    "backgroundColor": "black",
    "duration": 300,
    "easing": [Function],
    "opacity": 0.4,
    "useNativeDriver": true,
    "zIndex": 99,
  },
  "buttonDefaultProps": Object {
    "TouchableComponent": [Function],
    "defaultStyles": "object",
  },
  "buttonDefaultStyles": Object {
    "disabledText": Object {
      "color": "#cdcdcd",
    },
    "disabledView": Object {},
    "text": Object {
      "alignSelf": "center",
      "color": "#007aff",
      "fontSize": 18,
    },
    "view": Object {
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
    },
  },
  "inputButtonDefaultProps": Object {
    "placeholderTextColor": "#c7c7cc",
  },
  "inputButtonDefaultStyles": Object {
    "disabledText": Object {
      "color": "#cdcdcd",
    },
    "disabledView": Object {},
    "text": Object {
      "alignSelf": "flex-start",
      "color": "#007aff",
      "fontSize": 18,
    },
    "view": Object {
      "justifyContent": "flex-start",
    },
  },
  "pressableDefaultProps": Object {
    "TouchableComponent": [Function],
  },
  "pressableDefaultStyles": Object {
    "disabledText": Object {
      "color": "#cdcdcd",
    },
    "disabledView": Object {},
    "text": Object {
      "alignSelf": "center",
      "color": "#007aff",
      "fontSize": 18,
    },
    "view": Object {
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
    },
  },
}
`;
